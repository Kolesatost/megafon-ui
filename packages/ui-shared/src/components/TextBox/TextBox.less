@import '~@megafon/ui-core/styles/base.less';

@b: e('.mfui-text-box');

@{b} {
    &__content :first-child {
        margin-top: 0 !important;
    }

    &__content :last-child {
        margin-bottom: 0 !important;
    }

    &_text-center {
        text-align: center;
    }
}
