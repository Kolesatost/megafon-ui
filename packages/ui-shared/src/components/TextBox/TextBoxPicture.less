@import '~@megafon/ui-core/styles/base.less';

@b: e('.mfui-beta-text-box-picture');

@{b} {
    &_margin_default {
        margin: 16px 0;
    }

    &_margin_big_top_margin {
        margin: 24px 0 16px;
    }

    &_margin_big_bottom_margin {
        margin: 16px 0 24px;
    }

    &_margin_big_vertical_margin {
        margin: 24px 0;
    }

    &__img {
        width: 100%;
        height: 100%;
    }
}
