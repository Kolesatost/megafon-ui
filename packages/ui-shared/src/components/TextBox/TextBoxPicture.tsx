import * as React from 'react';
import { cnCreate } from '@megafon/ui-helpers';
import * as PropTypes from 'prop-types';
import './TextBoxPicture.less';

export const pictureMarginTypes = {
    DEFAULT: 'default',
    BIG_TOP: 'big-top',
    BIG_BOTTOM: 'big-bottom',
    BIG_VERTICAL: 'big-vertical',
} as const;

type PictureMarginTypesType = typeof pictureMarginTypes[keyof typeof pictureMarginTypes];

export interface ITextBoxPictureProps {
    /** URL картинки */
    url: string;
    /** Значение вертикальных отступов */
    margin?: PictureMarginTypesType;
}

const cn = cnCreate('mfui-beta-text-box-picture');
const TextBoxPicture: React.FC<ITextBoxPictureProps> = ({ url, margin = pictureMarginTypes.DEFAULT }) => (
    <div className={cn({ margin })}>
        <img className={cn('img')} src={url} alt="text-box" />
    </div>
);

TextBoxPicture.propTypes = {
    url: PropTypes.string.isRequired,
    margin: PropTypes.oneOf([
        pictureMarginTypes.DEFAULT,
        pictureMarginTypes.BIG_TOP,
        pictureMarginTypes.BIG_BOTTOM,
        pictureMarginTypes.BIG_VERTICAL,
    ]),
};

export default TextBoxPicture;
