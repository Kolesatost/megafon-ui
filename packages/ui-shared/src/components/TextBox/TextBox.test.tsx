import * as React from 'react';
import { mount, shallow } from 'enzyme';
import TextBox from './TextBox';

describe('TextBox', () => {
    it('render with default props', () => {
        const wrapper = shallow(
            <TextBox>
                <p>test</p>
            </TextBox>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('render with className="test-class-name"', () => {
        const wrapper = shallow(
            <TextBox className="test-class-name">
                <p>test</p>
            </TextBox>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('render with centered text', () => {
        const wrapper = shallow(
            <TextBox textCenter={true}>
                <p>test</p>
            </TextBox>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('render with limited width', () => {
        const wrapper = shallow(
            <TextBox isFullWidth={false}>
                <p>test</p>
            </TextBox>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('render with limited width and centered text', () => {
        const wrapper = shallow(
            <TextBox isFullWidth={false} textCenter={true}>
                <p>test</p>
            </TextBox>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('render with limited width and centered component', () => {
        const wrapper = shallow(
            <TextBox isFullWidth={false} centeringWithLimitedWidth={true}>
                <p>test</p>
            </TextBox>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('render with limited width, centered component and centered text', () => {
        const wrapper = shallow(
            <TextBox isFullWidth={false} centeringWithLimitedWidth={true} textCenter={true}>
                <p>test</p>
            </TextBox>,
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should return reference to root element', () => {
        const ref: React.RefObject<HTMLDivElement> = React.createRef();
        mount(
            <TextBox rootRef={ref}>
                <p>test</p>
            </TextBox>,
        );

        expect(ref.current).not.toBeNull();
    });
});
