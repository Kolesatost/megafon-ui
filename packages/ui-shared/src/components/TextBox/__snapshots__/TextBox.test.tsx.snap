// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextBox render with centered text 1`] = `
<div
  className="mfui-beta-text-box mfui-beta-text-box_text-center"
>
  <div
    className="mfui-beta-text-box__content"
  >
    <p>
      test
    </p>
  </div>
</div>
`;

exports[`TextBox render with className="test-class-name" 1`] = `
<div
  className="mfui-beta-text-box test-class-name"
>
  <div
    className="mfui-beta-text-box__content"
  >
    <p>
      test
    </p>
  </div>
</div>
`;

exports[`TextBox render with default props 1`] = `
<div
  className="mfui-beta-text-box"
>
  <div
    className="mfui-beta-text-box__content"
  >
    <p>
      test
    </p>
  </div>
</div>
`;

exports[`TextBox render with limited width 1`] = `
<div
  className="mfui-beta-text-box"
>
  <Grid
    hAlign="left"
  >
    <GridColumn
      desktop="10"
      mobile="12"
      tablet="12"
      wide="8"
    >
      <div
        className="mfui-beta-text-box__content"
      >
        <p>
          test
        </p>
      </div>
    </GridColumn>
  </Grid>
</div>
`;

exports[`TextBox render with limited width and centered component 1`] = `
<div
  className="mfui-beta-text-box"
>
  <Grid
    hAlign="center"
  >
    <GridColumn
      desktop="10"
      mobile="12"
      tablet="12"
      wide="8"
    >
      <div
        className="mfui-beta-text-box__content"
      >
        <p>
          test
        </p>
      </div>
    </GridColumn>
  </Grid>
</div>
`;

exports[`TextBox render with limited width and centered text 1`] = `
<div
  className="mfui-beta-text-box mfui-beta-text-box_text-center"
>
  <Grid
    hAlign="left"
  >
    <GridColumn
      desktop="10"
      mobile="12"
      tablet="12"
      wide="8"
    >
      <div
        className="mfui-beta-text-box__content"
      >
        <p>
          test
        </p>
      </div>
    </GridColumn>
  </Grid>
</div>
`;

exports[`TextBox render with limited width, centered component and centered text 1`] = `
<div
  className="mfui-beta-text-box mfui-beta-text-box_text-center"
>
  <Grid
    hAlign="center"
  >
    <GridColumn
      desktop="10"
      mobile="12"
      tablet="12"
      wide="8"
    >
      <div
        className="mfui-beta-text-box__content"
      >
        <p>
          test
        </p>
      </div>
    </GridColumn>
  </Grid>
</div>
`;
